code/turf/floors.dm

```diff
@@ -1544,6 +1548,65 @@ TYPEINFO(/turf/simulated/floor/grass)
 /turf/proc/grassify()
 	.=0
 
+/turf/proc/snow_prints(atom/movable/AM)
+	if (isliving(AM))
+		var/mob/living/M = AM
```

Comment:
recommend inverting the logic here to return early instead, and also return early if the mob is floating:

```
	if (!isliving(AM))
		return
	if (HAS_ATOM_PROPERTY(AM, PROP_ATOM_FLOATING))
		return
	var/mob/living/M = AM
	...
```

Reply:
Added floating check.